config/env.config.js

```diff
@@ -55,5 +55,60 @@ const config = {
     }
 };
 
+const RedirectPlugin = function(options) { 
+    this.apply = function(compiler) {
+        compiler.plugin('compilation', function(compilation) {
+            compilation.plugin('html-webpack-plugin-before-html-processing', function(htmlPluginData, callback) {
+                let htmlHead = htmlPluginData.html.match('<head>');
+                if (htmlHead) {
+                    htmlHead = htmlHead.index;
+                    htmlPluginData.html = htmlPluginData.html.slice(0, htmlHead) + '<head>' +
+                    `<script>
+                        var getParameterByName = function(name) {
+                            var url = window.location.search;
+                            var queryExp = new RegExp("[\\?&]"+name+"=([^&#]*)", "i");
+                            var match = queryExp.exec(url);
+                            if (match && match.length > 1) {
+                                return match[1];
+                            }
+                            return null;
+                        };
+                        var originUrl = getParameterByName("originEnvironment");
+                        var newUrl = getParameterByName("newEnvironment");
+                        if (newUrl) {
+                            newUrl = decodeURIComponent(newUrl);
+                            if (window.location.href.match(newUrl)) {
+                                window.localStorage.removeItem("environmentConfig");
+                            } else {
+                                window.localStorage.setItem("environmentConfig", newUrl);
+                            }
+                        }
+
+                        if (window.localStorage.getItem("environmentConfig")) {
+                            var originParam = "" + 
+                            (window.location.search ? "&" : "?") + 
```

Comment:
Indentation is odd here, as is below.  Also, what about using `[...].join()` instead?